firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -451,33 +516,34 @@ public void testGetAuthToken_multipleCallsDoNotForceRefresh_fetchedNewTokenOnce(
 
   @Test
   public void testGetAuthToken_multipleCallsForceRefresh_fetchedNewTokenTwice() throws Exception {
-    when(mockPersistedFid.readPersistedFidEntryValue()).thenReturn(REGISTERED_FID_ENTRY);
-    // Use a mock ServiceClient for network calls with delay(1000ms) to ensure first task is not
+    persistedFid.insertOrUpdatePersistedFidEntry(REGISTERED_FID_ENTRY);
+    // Use a mock ServiceClient for network calls with delay(500ms) to ensure first task is not
     // completed before the second task starts. Hence, we can test multiple calls to getAuthToken()
     // and verify one task waits for another task to complete.
 
     doAnswer(
             AdditionalAnswers.answersWithDelay(
-                1000,
+                500,
                 (unused) ->
                     InstallationTokenResult.builder()
                         .setToken(TEST_AUTH_TOKEN_3)
                         .setTokenExpirationInSecs(TEST_TOKEN_EXPIRATION_TIMESTAMP)
                         .build()))
         .doAnswer(
             AdditionalAnswers.answersWithDelay(
-                1000,
+                500,
                 (unused) ->
                     InstallationTokenResult.builder()
                         .setToken(TEST_AUTH_TOKEN_4)
                         .setTokenExpirationInSecs(TEST_TOKEN_EXPIRATION_TIMESTAMP)
                         .build()))
         .when(backendClientReturnsOk)
         .generateAuthToken(anyString(), anyString(), anyString(), anyString());
+    when(mockUtils.isAuthTokenExpired(any())).thenReturn(false);
```

Comment:
I skipped updating ```any()``` to a specific FID entry for ```mockUtils.isAuthTokenExpired()``` only here. This was to hide the details that don't improve the test. Hope that's fine.

Reply:
that's fine, I think the rest of the cases have benefited from the change